### PR TITLE
[MM-59367] export: enable exporting thread followers for CRT

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Decided to split into several PRs to ensure better readability and would be easier for all of us work on. This PR only includes the thread membership exports which is essential to ensure CRT properties are carried over for an export/import process.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59367

#### Release Note

```release-note
CRT memberships are now exportable for bulk export.
```

### CODE REVIEW

File: server/channels/app/imports/import_types.go [hunk 1]
```diff
@@ -236,3 +240,9 @@ type ComparablePreference struct {
 	Category string
 	Name     string
 }
+
+type ThreadFollowerImportData struct {
+	User           *string `json:"user"`
+	LastViewAt     *int64  `json:"last_view_at"`
+	UnreadMentions *int64  `json:"unread_mentions"`
```
Comment (agnivade) [inline server/channels/app/imports/import_types.go:247]: Do we need `omitempty` for these?

File: server/channels/store/storetest/thread_store.go [hunk 2]
```diff
@@ -1201,6 +1201,75 @@ func testVarious(t *testing.T, rctx request.CTX, ss store.Store) {
...
+				UpdateFollowing:       true,
+				UpdateViewedTimestamp: false,
+				UpdateParticipants:    true,
+			})
+			require.NoError(t, err)
+
+			members, err = ss.Thread().GetThreadMembershipsForExport(team1channel1post1.Id)
+			require.NoError(t, err)
+
+			// team1channel1post1 should have 1 followin member
```
Comment (agnivade) [inline server/channels/store/storetest/thread_store.go:1260]: typo at "following"

File: server/public/model/thread.go [hunk 3]
```diff
@@ -126,3 +126,9 @@ type ThreadMembership struct {
 	// threads with the mention count.
 	UnreadMentions int64 `json:"unread_mentions"`
 }
+
+type ThreadMembershipForExport struct {
+	UserName       string `json:"user_name"`
+	LastViewed     int64  `json:"last_view"`
```
Comment (agnivade) [inline server/public/model/thread.go:132]: typo "last_viewed"

**Review by agnivade — APPROVED** [review]
Just some minor comments. Nothing blocking

**Review by streamer45 — APPROVED** [review]
Looks good, nothing blocking. I added the QA label to have Dylan go through this work once everything else is merged.

File: server/channels/app/imports/import_types.go (hunk 1 above)
Comment (isacikgoz) [reply server/channels/app/imports/import_types.go:247]: Yeah, I think so. Good point 👍

File: server/channels/app/imports/import_types.go [hunk 4]
```diff
@@ -170,6 +170,8 @@ type PostImportData struct {
 	Replies     *[]ReplyImportData      `json:"replies,omitempty"`
 	Attachments *[]AttachmentImportData `json:"attachments,omitempty"`
 	IsPinned    *bool                   `json:"is_pinned,omitempty"`
+
+	ThreadFollowers *[]ThreadFollowerImportData `json:"thread_followers,omitempty"`
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/channels/app/imports/import_types.go:174]: It was mostly to keep it consistent really.

File: server/channels/app/imports/import_types.go [hunk 5]
```diff
@@ -236,3 +240,9 @@ type ComparablePreference struct {
 	Category string
 	Name     string
 }
+
+type ThreadFollowerImportData struct {
+	User           *string `json:"user"`
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/channels/app/imports/import_types.go:245]: I just wanted to keep it short on JSON file to be efficient on the file size and a consequence of being it coherent with the tag. And also keep it consistent with other entities such as PostImportData.

**Review by yasserfaraazkhan — APPROVED** [review]
Thank you for helping with QA steps Ibrahim

- On a local instance, enabled CRT and created threaded posts with replies from users.
- Using mmctl exported and downloaded the file.
- Cleaned the server and created a new user to login.
- Used mmct to import the data file
- Logged in to new server.
Successfully verified the CRTs are carried over with followers nad replies.

**Pull request merged**